testsuite/standalone/src/test/java/org/wildfly/core/test/standalone/mgmt/api/core/DeploymentModulesListTestCase.java

```diff
@@ -0,0 +1,279 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.core.test.standalone.mgmt.api.core;
+
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.Operation;
+import org.jboss.as.controller.client.helpers.ClientConstants;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.test.deployment.trivial.ServiceActivatorDeploymentUtil;
+import org.jboss.as.test.shared.PermissionUtils;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.Property;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.exporter.ZipExporter;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.core.testrunner.ManagementClient;
+import org.wildfly.core.testrunner.WildflyTestRunner;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.PropertyPermission;
+import java.util.stream.Collectors;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.CONTENT;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.ENABLED;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.LIST_MODULES;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP_ADDR;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.RESULT;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.VERBOSE;
+import static org.wildfly.common.Assert.assertFalse;
+import static org.wildfly.common.Assert.assertTrue;
+
+/**
+ * List modules which are on deployment’s classpath
+ * /deployment=application_war_ear_name:list-modules(verbose=false|true)
+ * @author <a href="mailto:szhantem@redhat.com">Sultan Zhantemirov</a> (c) 2019 Red Hat, inc.
+ */
+@RunWith(WildflyTestRunner.class)
+public class DeploymentModulesListTestCase {
+
+    private JavaArchive archive;
+    private static final String NODE_TYPE = "deployment";
+    private static final String EXAMPLE_MODULE_TO_EXCLUDE = "ibm.jdk";
+    private static final String EXAMPLE_USER_MODULE = "jdk.net";
+    private static final String JAR_DEPLOYMENT_NAME_SUFFIX = "-module-test.jar";
+    private static final String JAR_DEPLOYMENT_NAME = EXAMPLE_USER_MODULE + JAR_DEPLOYMENT_NAME_SUFFIX;
+
+    @Inject
+    private static ManagementClient managementClient;
+    private ModelControllerClient client;
+
+    @Before
```

Comment:
your code is running fine but I don't understand why you can't simply use `@Deployment` to create and deploy your archive?

Reply:
@jmesnil Hi Jeff, that was my first written WildFly deployment related test case. I wasn't aware of such annotation existence. I will have to use @Deployment in my future test cases.

Reply:
reason: we are using WildflyTestRunner for the test, so `@Deployment` is not available. We don't have arquillian libraries in core (I didn't know that), but it makes sense to use arquillian in wildfly-full since it is the standard way to do it. I'll comment the PR there.